src/routes/friend.js

```diff
@@ -1,88 +1,197 @@
+      })
+      .catch(err => res.status(400).json('Error: ' + err));
 })
 
-// send  a friend rerquest
-router.route('/add').post(async (req, res) => {
-  const username = req.body.username.toString();
-  const friendUsername = req.body.friendUsername.toString()
-  StudentRepository.findOneByUsername(username)
-    .then((myProfile) => {
-      StudentRepository.findOneByUsername(friendUsername).then((friendProfile) => {
-        let friend = friendUsername;
-        let me = username;
-
-        let friendRequestsSent = myProfile.friendRequestsSent
-        friendRequestsSent.push(friendUsername)
-
-        let friendRequestsReceived = friendProfile.friendRequestsReceived
-        friendRequestsReceived.push(username)
-
-        StudentRepository.updateFriendRequestSent(me, friendRequestsSent)
-          .then()
-          .catch(err => res.status(400).json('Error: ' + err));
-
-        StudentRepository.updateFriendRequestReceived(friend, friendRequestsReceived)
-          .then()
-          .catch(err => res.status(400).json('Error: ' + err));
-      })
-        .catch(err => res.status(400).json('Error: ' + err));
-    })
-    .catch(err => res.status(400).json('Error: ' + err));
-});
+/**
+ * @author: Jasmin Guay
+ * Endpoint to update a student friend list.
+ * @params email (string) email of the student to be updated
+ * @params friends (string[]) list of updated friend list.
+ * @return the update friend list.
+ */
+router.route('/updateFriendList').post( async (req,res) => {
+    const email = req.body.email.toString();
+    const updatedFriendList = req.body.friends.toString();
 
-router.route('/answer').post(async (req, res) => {
+    StudentRepository.updateFriendList(email,updatedFriendList)
+        .then((student) => { res.json(student.friends).status(200) })
+        .catch((err) => { res.json(`Error happened in updateFriendList - ${err}`).status(404) })
+})
 
-  const username = req.body.username.toString();
-  const friendUsername = req.body.friendUsername
-  const answer = req.body.answer;
 
-  Student.updateOne({ username: username },
-    { $pull: { friendRequestsReceived: { $in: [friendUsername] } } })
+/**
+ * @author: Jasmin Guay
+ * Endpoint to answer a friend request.
+ * @param requestId (string) id of the request.
+ * @param answer ("accepted" | "declined") answer to the friend request.
+ * @return studentUsername (string) username of the student added to the friend list if the request was accepted, null otherwise.
+ */
+router.route('/answerFriendRequest').post(async (req, res) => {
+  const requestId = req.body.requestId.toString();
+  const request = await FriendRequest.findOne({_id: requestId});
 
+  if (request == null) {
+    res.json(`No request found with id [${requestId}`).status(404);
+  } else {
+    if (req.body.answer === "accepted") {
+      await addToFriendLists(request.senderEmail, request.receiverEmail);
+      res.json(request.receiverEmail).status(200);
+    } else {
```

Comment:
For this else, as privately discussed, it would ideally not give a 200 OK code and delete the request (for cases where someone maliciously hits the backend with a valid request id). Although as you mentioned, once we implement user tokens on these routes, this should not be a concern. It should be noted, however, that we do not verify which user is making the request and whether the request is theirs. So hypothetically, if I have an account (and consequently a token), I can hypothetically accept someone else's friend request if I have their request if (even though they did not accept the request themselves). An issue will be raised for this, so for now, the PR can be approved.